lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -2,7 +2,9 @@
 module KubernetesDeploy
   class Pod < KubernetesResource
     TIMEOUT = 10.minutes
-
+    PREDEPLOY = true
+    PREDEPLOY_DEPENDENCIES = %w(ResourceQuota ServiceAccount ConfigMap PersistentVolumeClaim)
```

Comment:
Extracted from the previous static predeploy dependency list. Note that this is a partial order (the previous list by definition was a total order).

Reply:
What about the fact that this can depend on custom resources, such as cloudsql/redis?

Reply:
Then we'll need some way to specify such custom dependencies in the pod spec.

Alternatively we could optionally specify something like `PREDEPLOY_DEPENDENTS` on resources, so for e.g. `CloudSQL` could specify that it is a dependency of `Pod` which is its dependent.